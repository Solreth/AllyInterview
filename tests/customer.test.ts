import graphQLRequest from "../src/utils/graphRequest";

const customerTestObj = [
    {
    gender: expect.any(String),
    name: {
        title: expect.any(String),
        first: expect.any(String),
        last: expect.any(String)
    },
    email: expect.any(String),
    phone: expect.any(String),
    }
];

it('should return an array of customers', async () => {
    const customers = await graphQLRequest(`
    query Customers {
        customers {
            gender
            name {
                title
                first
                last
            }
            email
            phone
        }
      }
    `)
    expect(customers.body.data.customers).toMatchObject(customerTestObj);
})