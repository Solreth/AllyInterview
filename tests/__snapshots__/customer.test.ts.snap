// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return an array of customers 1`] = `
Array [
  Object {
    "email": "rick.nelson@example.com",
    "gender": "male",
    "name": Object {
      "first": "Rick",
    },
    "phone": "07-5547-3385",
  },
]
`;
